### PR TITLE
Backport "HBASE-27903 Skip submitting Split/Merge procedure when split/merge is disabled at table level" to branch-2

### PR DESCRIPTION
**Pull request by NihalJain**
Backports #6169

### CODE REVIEW

**Comment by NihalJain** [comment]
This did not apply cleanly, will first merge this backport and then backport this to older branches.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 42s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  6s |  |  branch-2 passed  |
| +1 :green_heart: |  compile  |   2m 49s |  |  branch-2 passed  |
| +1 :green_heart: |  checkstyle  |   0m 39s |  |  branch-2 passed  |
| +1 :green_heart: |  spotbugs  |   1m 36s |  |  branch-2 passed  |
| +1 :green_heart: |  spotless  |   0m 47s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 54s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 53s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 53s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 40s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m 40s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  15m 33s |  |  Patch does not cause any errors with Hadoop 2.10.2 or 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 45s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 13s |  |  The patch does not generate ASF License warnings.  |
|  |   |  36m  3s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6186 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 1d7bd4ae1a4a 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / a55fbf2223fdc76791268dccf450f037ebb66417 |
| Default Java | Eclipse Adoptium-11.0.23+9 |
| Max. process+thread count | 77 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 19s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  5s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  2s |  |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 57s |  |  branch-2 passed  |
| +1 :green_heart: |  javadoc  |   0m 30s |  |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   5m 31s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 53s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 30s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 30s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 213m 36s | [/patch-unit-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-hbase-server.txt) |  hbase-server in the patch failed.  |
|  |   | 238m 28s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6186 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 89185174a9c0 5.4.0-192-generic #212-Ubuntu SMP Fri Jul 5 09:47:39 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / a55fbf2223fdc76791268dccf450f037ebb66417 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/testReport/ |
| Max. process+thread count | 4860 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by NihalJain** [comment]
Failures are related, need to see in branch-2 why exception type is not coming as expected, even hough same code change

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 40s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  6s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 34s |  |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 45s |  |  branch-2 passed  |
| +1 :green_heart: |  javadoc  |   0m 26s |  |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   4m 56s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 28s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 44s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 44s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 26s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 58s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 233m  8s | [/patch-unit-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/artifact/yetus-jdk8-hadoop2-check/output/patch-unit-hbase-server.txt) |  hbase-server in the patch failed.  |
|  |   | 255m 43s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6186 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux ce4aae8ee39b 5.4.0-192-generic #212-Ubuntu SMP Fri Jul 5 09:47:39 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / a55fbf2223fdc76791268dccf450f037ebb66417 |
| Default Java | Temurin-1.8.0_412-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/testReport/ |
| Max. process+thread count | 4103 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 40s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  6s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  5s |  |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 56s |  |  branch-2 passed  |
| +1 :green_heart: |  javadoc  |   0m 29s |  |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   5m 35s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  3s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 27s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 36s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 263m 36s | [/patch-unit-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt) |  hbase-server in the patch failed.  |
|  |   | 289m 15s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6186 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 2089c5742e6d 5.4.0-192-generic #212-Ubuntu SMP Fri Jul 5 09:47:39 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / a55fbf2223fdc76791268dccf450f037ebb66417 |
| Default Java | Eclipse Adoptium-11.0.23+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/testReport/ |
| Max. process+thread count | 4495 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 40s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  5s |  |  branch-2 passed  |
| +1 :green_heart: |  compile  |   2m 49s |  |  branch-2 passed  |
| +1 :green_heart: |  checkstyle  |   0m 40s |  |  branch-2 passed  |
| +1 :green_heart: |  spotbugs  |   1m 34s |  |  branch-2 passed  |
| +1 :green_heart: |  spotless  |   0m 47s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 54s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 50s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 50s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 39s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m 42s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  15m 34s |  |  Patch does not cause any errors with Hadoop 2.10.2 or 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 45s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 12s |  |  The patch does not generate ASF License warnings.  |
|  |   |  35m 56s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6186 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 70bab4c14162 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 1239f511a928a03bdeec0de9f97bcfb67f4cb840 |
| Default Java | Eclipse Adoptium-11.0.23+9 |
| Max. process+thread count | 79 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 44s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  5s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ branch-2 Compile Tests _ |
| -1 :x: |  mvninstall  |   0m 47s | [/branch-mvninstall-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/artifact/yetus-jdk8-hadoop2-check/output/branch-mvninstall-root.txt) |  root in branch-2 failed.  |
| -1 :x: |  compile  |   0m 11s | [/branch-compile-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/artifact/yetus-jdk8-hadoop2-check/output/branch-compile-hbase-server.txt) |  hbase-server in branch-2 failed.  |
| -0 :warning: |  javadoc  |   0m  7s | [/branch-javadoc-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/artifact/yetus-jdk8-hadoop2-check/output/branch-javadoc-hbase-server.txt) |  hbase-server in branch-2 failed.  |
| +1 :green_heart: |  shadedjars  |   5m 19s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 25s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 42s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 42s |  |  the patch passed  |
| -0 :warning: |  javadoc  |   0m 24s | [/results-javadoc-javadoc-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/artifact/yetus-jdk8-hadoop2-check/output/results-javadoc-javadoc-hbase-server.txt) |  hbase-server generated 21 new + 0 unchanged - 0 fixed = 21 total (was 0)  |
| +1 :green_heart: |  shadedjars  |   5m 20s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 216m  6s |  |  hbase-server in the patch passed.  |
|  |   | 236m 30s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6186 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 733e6a8679e3 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 1239f511a928a03bdeec0de9f97bcfb67f4cb840 |
| Default Java | Temurin-1.8.0_412-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/testReport/ |
| Max. process+thread count | 4342 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 41s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  5s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  8s |  |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 55s |  |  branch-2 passed  |
| +1 :green_heart: |  javadoc  |   0m 29s |  |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   5m 37s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  8s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 28s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 39s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 235m 21s |  |  hbase-server in the patch passed.  |
|  |   | 261m  7s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6186 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux ee7061ff0986 5.4.0-192-generic #212-Ubuntu SMP Fri Jul 5 09:47:39 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 1239f511a928a03bdeec0de9f97bcfb67f4cb840 |
| Default Java | Eclipse Adoptium-11.0.23+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/testReport/ |
| Max. process+thread count | 4530 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 47s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  6s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  6s |  |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 58s |  |  branch-2 passed  |
| +1 :green_heart: |  javadoc  |   0m 30s |  |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   5m 31s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  0s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 58s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 58s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 30s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 31s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 266m  5s | [/patch-unit-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-hbase-server.txt) |  hbase-server in the patch failed.  |
|  |   | 291m 42s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6186 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux d9350a4cddb0 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 1239f511a928a03bdeec0de9f97bcfb67f4cb840 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/testReport/ |
| Max. process+thread count | 4249 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6186/2/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestSplitOrMergeAtTableLevel.java [hunk 1]
```diff
@@ -157,15 +157,17 @@ private void trySplitAndEnsureItFails(final TableName tableName) throws Exceptio
     List<RegionInfo> regions = admin.getRegions(tableName);
     int originalCount = regions.size();
 
-    // split the table and make sure region count does not increase
-    Future<?> f = admin.splitRegionAsync(regions.get(0).getEncodedNameAsBytes(), Bytes.toBytes(2));
     try {
+      // split the table and make sure region count does not increase
+      Future<?> f =
```
Comment (NihalJain) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestSplitOrMergeAtTableLevel.java:162]: Had to modify tests as in branch-2, the HBTU creates an instance of HBaseAdmin whose methods splitRegionAsync/mergeRegionsAsync seem to propagate error immediately without a call to future.get() which is in contrast to how the admin instance of HBTU works in 3.x.

CC: @Apache9

Comment (NihalJain) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestSplitOrMergeAtTableLevel.java:162]: Gentle ping @Apache9 are we good with this?

Comment (Apache9) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestSplitOrMergeAtTableLevel.java:162]: There are some behavior changes when reimplementing sync client on top of async client. I think it is OK.

**Review by Apache9 — APPROVED** [review]

File: hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestSplitOrMergeAtTableLevel.java (hunk 1 above)
Comment (NihalJain) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestSplitOrMergeAtTableLevel.java:162]: Thank you @Apache9 for the heads up.

**Pull request merged**